### DISCUSSION TITLE
v3.2.2 - fixed some bugs performance improvement

### DISCUSSION THREAD

**Discussion by Blackymas** [Blackymas/NSPanel_HA_Blueprint#595] [Announcements]
### As always, we would like to thank everyone who donated to this project and remind you that every donation helps to support this project - so if you haven't made a donation yet, now would be a good time ;)

[![Paypal](https://user-images.githubusercontent.com/41958506/212499642-b2fd097a-0938-4bfc-b37b-74df64592c58.png)](https://www.paypal.com/donate/?hosted_button_id=S974SWQMB8PB2)

&nbsp;

### Additionally, there are some errors that we have fixed.

&nbsp;

### Please update the ESPhome file first before updating the TFT for this update. If the TFT is updated first, it may cause problems!

&nbsp;

### The full release notes of v3.2 can be found here: https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v.3.2

&nbsp;

## Files that need to be reloaded:

1. nspanel_esphome.yaml - v3.2.2
2. nspanel.tft - v3.2.2
3. nspanel_blueprint.yaml - v3.2.2

> ⚠️ Please update the ESPhome file first before updating the TFT 

&nbsp;

 _Update procedure can be read here:_

EN: https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-HowTo---All-important-thing-you-should-know

DE: https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(DE)-HowTo---Alle-wichtigen-Dinge-die-man-wissen-sollte

&nbsp;

## Community Updates

&nbsp;

## Overview of all changes

1. **_"Improved Button Page Performance"_** - Now, Buttonpages 01-04 load faster.

2. **_"Fix for "blue lines" when fallback was enabled"_** - Graphics were missing when fallback was activated.

3. **_"ignore unassigned indoor sensor "_** - When no indoor sensor was set, the display remained stuck at loading.

4. **_"Fix ESPhome problem"_** - In some cases, an error message occurred due to "external_components". This could also fix the issue with TFT upload.

5. **_"Fix sleep mode"_** - Fixed issue with waking up the display in sleep mode.

6. **_"Fix weather units"_** - Fixed issue where weather pages display incorrect units. Now it will display the units provided by the Weather integration, when available.
&nbsp;

## Details of all changes

&nbsp;

## Next topics we are currently working on
See here: https://github.com/Blackymas/NSPanel_HA_Blueprint/labels/roadmap

<hr /><em>This discussion was created from the release <a href='https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v3.2.2'>v3.2.2 - fixed some bugs performance improvement</a>.</em>

**Comment by semml0r** [comment]
Thanks a lot for all your efforts! Update works without issues after restarting HA :)

**Comment by scoudibou** [comment]
Update OK!
The loading time is much better and more stable ~1 sec to change pages..
Thank you !

**Comment by hkr086** [comment]
Thank you thank you.. now it's perfect!

**Comment by thesebastianf** [comment]
I updated from 3.1.0 to 3.2.2 and now it does not work so flawlessly like before for me.
So maybe somebody has a hint what I could improve.

When I boot the NSPanel before it did show all versions as 3.1.0
For Blueprint this is now blank (I updated the blueprint to 3.2.2 of course)
![image](https://user-images.githubusercontent.com/13987575/230388342-30ffddbf-1122-4f4e-8ec4-d544ed3f5a48.png)

And after that it does show "loading..." forever
![image](https://user-images.githubusercontent.com/13987575/230389388-1f110733-7bfd-42f0-8afa-2ca2557e3ded.png)



the correct UI is shown only after I go to "automation" in homeassistant change just anything in the config for the NSPanel and click SAVE. From there on it does work. This was definately different with 3.1.0 where it just worked also after a reboot of the NSPanel without any manual interaction.

**Reply by semml0r** [reply]
hi, I am just another user. Did you try restarting / rebooting Homeassistant? I faced the exact same, then restarted HA and it worked.

**Reply by thesebastianf** [reply]
yes I restarted homeassistant at least twice since then but it is still happening.

**Reply by semml0r** [reply]
did you try rebooting the nspanel from HA?

**Reply by thesebastianf** [reply]
just tried this.

- Reboot from HA --> still shows blueprint: blank on reboot. But does actually go to the UI Homepage 

- Reboot the NSPanel (Power Off On cycle): Blueprint blank and endless "loading".

**Reply by edwardtfn** [reply]
@thesebastianf, could you please take a look at your Home Assistant logs and share the error messages related to the panel?

**Comment by Njamotep** [comment]
Thanks for all the hard work! Big improvement in loading time. Compared to 3.2.1 this is lightning fast! This I can live with 😃 !

**Comment by jjnj023** [comment]
Great work! Much snappier now!
Every time I use the panels, I get a smile on my face.

**Comment by Statusbit** [comment]
After update one of three panels have a mistake...

<img width="323" alt="Screenshot 2023-04-13 214859" src="https://user-images.githubusercontent.com/85311570/231869553-845e50c4-b2c0-429f-89bb-8993010c770f.png">


`Warum kann ich hier keinen kopierten Text platzieren?`

`NSPanel Wohnzimmer verwendet einen unbekannten Dienst
Die Automatisierung "NSPanel Wohnzimmer" (automation.nspanel_wohnzimmer) hat eine Aktion, die einen unbekannten Dienst aufruft: esphome.nspwohnzimmer_set_button.

Dieser Fehler verhindert, dass die Automatisierung ordnungsgemäß ausgeführt wird. Vielleicht ist dieser Dienst nicht mehr verfügbar oder vielleicht hat ein Tippfehler ihn verursacht.

Um diesen Fehler zu beheben, [bearbeite die Automatisierung](http://192.168.1.63:8123/config/automation/edit/1680437114816) und entferne die Aktion, die diesen Dienst aufruft.

Drücke unten auf SENDEN, um zu bestätigen, dass du diese Automatisierung korrigiert hast.`

**Reply by Statusbit** [reply]
Now it occurred to me that another automation sends a message to the display, has anything changed?

<img width="580" alt="1111Screenshot 2023-04-13 220251" src="https://user-images.githubusercontent.com/85311570/231870986-a7ecaf28-8100-4fe8-9b8d-00fba5b0a145.png">

**Reply by Statusbit** [reply]
At the panel comes "unknown message" Therefore I thought it is because of the messages. But probably he does not like that here is a lamp connected?

<img width="644" alt="222Screenshot 2023-04-13 221225" src="https://user-images.githubusercontent.com/85311570/231873173-4323611d-ef50-4b2d-929d-e660d53b847d.png">

**Reply by edwardtfn** [reply]
> After update one of three panels have a mistake...
> 
> <img alt="Screenshot 2023-04-13 214859" width="323" src="https://user-images.githubusercontent.com/85311570/231869553-845e50c4-b2c0-429f-89bb-8993010c770f.png">
> 
> `Warum kann ich hier keinen kopierten Text platzieren?`
> 
> `NSPanel Wohnzimmer verwendet einen unbekannten Dienst Die Automatisierung "NSPanel Wohnzimmer" (automation.nspanel_wohnzimmer) hat eine Aktion, die einen unbekannten Dienst aufruft: esphome.nspwohnzimmer_set_button.
> 
> Dieser Fehler verhindert, dass die Automatisierung ordnungsgemäß ausgeführt wird. Vielleicht ist dieser Dienst nicht mehr verfügbar oder vielleicht hat ein Tippfehler ihn verursacht.
> 
> Um diesen Fehler zu beheben, [bearbeite die Automatisierung](http://192.168.1.63:8123/config/automation/edit/1680437114816) und entferne die Aktion, die diesen Dienst aufruft.
> 
> Drücke unten auf SENDEN, um zu bestätigen, dass du diese Automatisierung korrigiert hast.`

This happens when your blueprint is updated before the ESPHome as the Blueprint will try to call a service which was not registered by the panel yet.
After updating everything (blueprint + esphome + TFT), wait a couple of minutes and the new services should be registered, then you can click at those messages and Mark as resolved.
If you still getting those messages back, try restarting your Home Assistant.

**Reply by edwardtfn** [reply]
> Now it occurred to me that another automation sends a message to the display, has anything changed?
> 
> <img alt="1111Screenshot 2023-04-13 220251" width="580" src="https://user-images.githubusercontent.com/85311570/231870986-a7ecaf28-8100-4fe8-9b8d-00fba5b0a145.png">

No, as far as I know this service haven't changed, but looks like you have changed your panel name in ESPHome. If that is the case, you most likely will get stuck with "Loading..." screen and, if this happens, try to rename you nspanel back to the previous name (or setup everything using the new name). I have added some coments about this on #671.

**Reply by Statusbit** [reply]
Thank you, works with patience.
